src/twisted/mail/mail.py

```diff
@@ -533,8 +533,7 @@ def addDomain(self, name, domain):
         @type domain: L{IDomain} provider
         @param domain: A domain object.
         """
-        portal = Portal(domain)
-        map(portal.registerChecker, domain.getCredentialsCheckers())
+        portal = Portal(domain, domain.getCredentialsCheckers())
```

Comment:
probably this code is (was) important

```suggestion
        portal = Portal(domain, domain.getCredentialsCheckers())
        for credentialsChecker in domain.getCredentialsCheckers():
            portal.registerChecker(credentialsChecker)
```

Reply:
this test would fail without this - if it were not skipped

https://github.com/twisted/twisted/blob/0315f0b05740597d9efbac4bde4cdcce779f8794/src/twisted/mail/test/test_mail.py#L983-L987

Reply:
Can we unskip it in *this* branch though? If so, great, if not, then we might just need to land #11898